### PR TITLE
Add missing evidence sources

### PR DESCRIPTION
**Pull request by xwatkins**
While links to Publications, UniProtKB and Protsite-Prorule were taken into account in evidence tags, lots of cases were missing. Jie provided the list of evidence tag sources with their corresponding urls. Even though there's overlap, these need to be kept separate for now because of the way they will be generated by the backend.

Most cases apart from references/publications are now handled in a more generic way.

### CODE REVIEW

File: src/components/UniProtEvidenceTag.tsx [hunk 1]
```diff
@@ -20,51 +21,50 @@ enum evidenceTagSourceTypes {
...
-  const groupedReferences =
-    references &&
-    groupBy(references, (reference: Evidence) => reference.source);
-  // TODO it looks like there's more source types than defined here
+  const groupedEvidences =
+    evidences && groupBy(evidences, (evidence: Evidence) => evidence.source);
+
+  const publicationReferences = groupedEvidences[evidenceTagSourceTypes.PUBMED];
+
+  delete groupedEvidences[evidenceTagSourceTypes.PUBMED];
```
[the remark this answers is not in the document]
Comment (xwatkins) [reply src/components/UniProtEvidenceTag.tsx]: but then the linter complains about `'groupedEvidencesWithoutPubs' is declared but its value is never read.`

[the remark this answers is not in the document]
Comment (dlrice) [reply src/components/UniProtEvidenceTag.tsx]: You can then use `groupedEvidencesWithoutPubs` at L48 in place of `groupedEvidences`:

```
{Object.keys(groupedEvidences).map(key => (
```

[the remark this answers is not in the document]
Comment (xwatkins) [reply src/components/UniProtEvidenceTag.tsx]: You're right, that's much nicer

**Comment by xwatkins** [comment]
Should all be addressed.

**Pull request merged**